bx_py_utils/anonymize.py

```diff
@@ -36,3 +36,20 @@ def anonymize(value: str, handle_email: bool = True) -> str:
 
     value = f'{value[:1]}{value[1:-1].translate(_ANONYMIZATION_TRANS)}{value[-1:]}'
     return value
+
+
+def anonymize_dict(data: dict, keys=('secret', 'password', 'token')) -> dict:
+    """
+    Returns a new dict with anonymized values for keys containing one of the given keywords.
+
+    >>> anonymize_dict({'client_id': '123', 'client_secret': 'This is really secret'})
+    {'client_id': '123', 'client_secret': 'Txxx_xx_xxxxxx_xxxxxt'}
+    """
+    anonymized_data = data.copy()  # Don't modify the original data!
+    for key, value in anonymized_data.items():
+        if value:
+            if isinstance(value, str) and any(k in key.lower() for k in keys):
```

Comment:
For the general case, can we require `keys` to be a (frozen)set or similar of lower-cased keys? Then this can be simplified to `if k in secret_keys and isinstance(value, str):`

Reply:
I lower the key in the source dict, not in the given secret_keys.
And we check that a part must only be matched, e.g: "ClientSecret" will match by "secret"

So i see no optimize potential here.

Reply:
Oh woops, my bad.